choose-index.md

```diff
@@ -195,197 +195,406 @@ mysql> EXPLAIN SELECT /*+ use_index_merge(t1, idx) */ * FROM t1 WHERE JSON_OVERL
+    +-------------------------------+---------+-----------+-----------------------------------------------------------------------------+---------------------------------------------+
+    
+    -- json_length(j->'$.a') = 3 cannot be accessed with IndexMerge directly, so TiDB uses IndexMerge to access the other two json_contains conditions, and json_length(j->'$.a') = 3 becomes a Selection operator.
+    > EXPLAIN SELECT /*+ use_index_merge(t4, mvi1) */ * FROM t4 WHERE json_contains(j->'$.a', '[1, 2]') AND json_contains(j->'$.a', '[3, 4]') AND json_length(j->'$.a') = 2;
+    +-------------------------------+---------+-----------+-----------------------------------------------------------------------------+----------------------------------------------------+
+    | id                            | estRows | task      | access object                                                               | operator info                                      |
+    +-------------------------------+---------+-----------+-----------------------------------------------------------------------------+----------------------------------------------------+
+    | IndexMerge_11                 | 0.00    | root      |                                                                             | type: intersection                                 |
+    | ├─IndexRangeScan_5(Build)     | 10.00   | cop[tikv] | table:t4, index:mvi1(cast(json_extract(`j`, _utf8'$.a') as unsigned array)) | range:[1,1], keep order:false, stats:pseudo        |
+    | ├─IndexRangeScan_6(Build)     | 10.00   | cop[tikv] | table:t4, index:mvi1(cast(json_extract(`j`, _utf8'$.a') as unsigned array)) | range:[2,2], keep order:false, stats:pseudo        |
+    | ├─IndexRangeScan_7(Build)     | 10.00   | cop[tikv] | table:t4, index:mvi1(cast(json_extract(`j`, _utf8'$.a') as unsigned array)) | range:[3,3], keep order:false, stats:pseudo        |
+    | ├─IndexRangeScan_8(Build)     | 10.00   | cop[tikv] | table:t4, index:mvi1(cast(json_extract(`j`, _utf8'$.a') as unsigned array)) | range:[4,4], keep order:false, stats:pseudo        |
+    | └─Selection_10(Probe)         | 0.00    | cop[tikv] |                                                                             | eq(json_length(json_extract(test.t4.j, "$.a")), 2) |
+    |   └─TableRowIDScan_9          | 0.00    | cop[tikv] | table:t4                                                                    | keep order:false, stats:pseudo                     |
+    +-------------------------------+---------+-----------+-----------------------------------------------------------------------------+----------------------------------------------------+
+    ```
+
+- All the conditions that are used for the IndexMerge must match the semantic of `OR` or `AND` that connects them.
```

Comment:
`semantic` should be replaced with `semantics` through the doc.

Reply:
Fixed